packages/ipfs-cpinner-provider/src/IpfsPinner.ts

```diff
@@ -0,0 +1,37 @@
+import { Repository } from 'typeorm'
+import { IpfsPinner } from './types'
+import { IpfsHttpClient } from 'ipfs-http-client'
+import IpfsPinnedCid from './entities/ipfs-pinned-cid'
+
+export default class implements IpfsPinner {
+  constructor (
+    private ipfsHttpClient: IpfsHttpClient,
+    private repository: Repository<IpfsPinnedCid>
+  ) {}
+
+  async pin (cid: string): Promise<boolean> {
+    const pinned = await this.repository.findOne({ where: { cid } })
+
+    if (pinned && pinned.count) {
```

Comment:
`!!0` evaluates false. This is not counting... We need to add a test for the case of deleting other's content.

Reply:
BTW it is not nullable, it has default value 1.